share/etest/runners.sh

```diff
@@ -108,6 +108,15 @@ run_single_test()
                 suite_setup
             fi
 
+            # Register __suite_teardown function as a trap callback and trigger it when the test receives an interrupt.
+            # If running the last test case, instead of triggering it by interrupt, make this function always triggered
+            # at the end of the test. 
+            if [[ ${testidx} -eq ${testidx_total} ]]; then
+                trap_add __suite_teardown EXIT
+            else
+                trap_add __suite_teardown ${DIE_SIGNALS[@]}
```

Comment:
The signal is trigger by either user's interrupt or parent process that sends the signal to this process when `failfast` is set.

Reply:
This is clever. Using `trap_add` to ensure that the suite teardown function is called is a really elegant solution to this.